### PR TITLE
Improve class overview UI

### LINKED ISSUE #1153: [UI idea] Time format of last-login of student is unclear
**Idea incl level**
The current format for the timestamp of the last login of a student is unclear. Being shown to the teacher as `2021-10-13T22:38:09`. It would be nice to adjust this to a more readable format such as `mm-hh dd-mm-yyyy` when it is a day **different** then today. It it's today it might be nice to shown just the time as `mm-hh today`.

### LINKED ISSUE #1154: [UI idea] Improve class UI to align with other pages
**Idea incl level**
Currently the UI of the class overview is represented as an un-styled table containing student information. It would be nice to improve this UI to better align with the rest of the website. Corresponding to this it might also be nice to give the teacher some additional information such as: total amount of students, highest level distribution and on the long-run: students that are having trouble to get along and should get extra attention.

**Additional context: Current UI**
Schermafbeelding 2021-11-03 om 12 02 37

### PR DESCRIPTION
**Pull request by TiBiBa**
Provide a general summary of your changes in the Title of the PR

**Description**
We improve the class overview UI to better align with the UI of the rest of the website. The table is styled and borders are adjusted to better differentiate between different sections. The header titles are shortened in the *.yaml files to better fit the UI and still contain the same message. The styling of the timestamp is improved through some additional pre-processing on the `isoformat()`.

**Fix for**

This PR fixed #1153 and fixed #1154. Originally, the purpose was to only fix #1154 but as they are very much aligned both are solved at once in this PR.

**How to test**

Make sure you have a teacher's account and navigate to one of your classes and notice the difference UI of the class overview.

**Screenshot**
Schermafbeelding 2021-11-15 om 13 11 51

Schermafbeelding 2021-11-15 om 13 12 17

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] Links to an existing issue or discussion (if not, create an issue first)
- [ ] Describes changes clear in the format above (present tense, no subject)
- [ ] Has a "how to test" section
- [ ] Only one thing is done in this pull request (specifically please try to refrain from mixing textual changes to the yamls from code changes)

### CODE REVIEW

**Comment by Felienne** [comment]
Looks good @TiBiBa!

I was thinking maybe only to change the text underneath the header "Leerling verwijderen", because now it will be a long list of the same text, which might be hard to oversee. What if we change that into "verwijder <username>"?

And maybe we can do the same for programs, instead of a long list of programs, make it "<username>'s programs"

**Review by Felienne — CHANGES_REQUESTED** [review]
See discussion

**Comment by TiBiBa** [comment]
> Looks good @TiBiBa!
> 
> 
> 
> I was thinking maybe only to change the text underneath the header "Leerling verwijderen", because now it will be a long list of the same text, which might be hard to oversee. What if we change that into "verwijder <username>"?
> 
> 
> 
> And maybe we can do the same for programs, instead of a long list of programs, make it "<username>'s programs"
> 
> 
> 
> 
> 
> 

I will take a look! We will have to add some additional values to the *.yaml files, but that shouldn't be a problem.

**Comment by Felienne** [comment]
> > Looks good @TiBiBa!
> > I was thinking maybe only to change the text underneath the header "Leerling verwijderen", because now it will be a long list of the same text, which might be hard to oversee. What if we change that into "verwijder "?
> > And maybe we can do the same for programs, instead of a long list of programs, make it "'s programs"
> 
> I will take a look! We will have to add some additional values to the *.yaml files, but that shouldn't be a problem.

Yeah that should not be a problem, we already have the headers so it is a matter of adding some words from there.

**Comment by TiBiBa** [comment]
I was a bit struggling with the next texts, but made some fixes with the last commit!

**Comment by Felienne** [comment]
Looks perfect for the programs, but for the student, I'd also like the name (so "Hedy verwijderen), is that possible? (I guess it is not hard since we have the name for the other column)

image

**Comment by TiBiBa** [comment]
> Looks perfect for the programs, but for the student, I'd also like the name (so "Hedy verwijderen"), is that possible? (I guess it is not hard since we have the name for the other column)
> 
> image

Yes that should be easy, will make a fix right away!

File: templates/class-overview.html [hunk 1]
```diff
@@ -23,9 +23,9 @@ <h2>{{auth.class_overview}}: {{class_info.name}}</h2>
             <td>{{student.last_login}}</td>
             <td>{{student.highest_level}}</td>
             <td>{{student.programs}}</td>
-            <td><a href="{{localize_link ('/programs?user=' + student.username)}}">{{auth.programs}}</a></td>
+            <td><a href="{{localize_link ('/programs?user=' + student.username)}}">{{student.username}}'s {{auth.page}}</a></td>
             <td>{% if student.latest_shared %}<a target="_blank" href="{{student.latest_shared.link}}">{{student.latest_shared.name}} ({{student.latest_shared.level}})</a>{% else %}-{% endif %}</td>
-            <td><a href="#" onclick="hedyApp.remove_student ('{{class_info.id}}', '{{student.username}}')">{{auth.remove_student}}</a></td>
+            <td><a href="#" onclick="hedyApp.remove_student ('{{class_info.id}}', '{{student.username}}')">{{student.username}} {{auth.remove}}</a></td>
```
Comment (Felienne) [inline templates/class-overview.html:28]: haha this is an interesting localization issue! For Dutch "Hedy verwijderen" makes sense but in En you would het "Hedy remove" which is weird and I do not know the proper order for other langs. I guess swapping the order and making it into _gebiedende wijs_: "remove Hedy/verwijder Hedy" is better?

**Review by Felienne — CHANGES_REQUESTED** [review]

File: templates/class-overview.html (hunk 1 above)
Comment (TiBiBa) [reply templates/class-overview.html]: Should be fixed with the last commit!

**Review by Felienne — APPROVED** [review]

**Comment by Felienne** [comment]
Looks good, we just have a conflict on class-overview that I do not know how to resolve.

**Comment by TiBiBa** [comment]
I just fixed the merge conflict, but I does not seem to auto-merge because a review is still missing

**Comment by Felienne** [comment]
@TiBiBa 

It seems this PR suffers from the same issues as the one with duplicate class names that is overwrites grammar and python changes, can you cherry pick the relevant things in this (or a fresh) PR?

**Comment by Felienne** [comment]
This seems to be the offender that needs to be removed: https://github.com/Felienne/hedy/pull/1234/commits/10c2082085825e80ebea8fe885f4c97e32518044

**Comment by TiBiBa** [comment]
> @TiBiBa
> 
> It seems this PR suffers from the same issues as the one with duplicate class names that is overwrites grammar and python changes, can you cherry pick the relevant things in this (or a fresh) PR?

I will make a new PR with a new branch to be sure that it all works well.

**Comment by TiBiBa** [comment]
This has become a chaos, a new PR will be made with same changes.

**Pull request closed without merging**